tests/difference/core/quint_model/ccv_test.qnt

```diff
@@ -0,0 +1,255 @@
+    }
+
+    // add a packet on the consumer
+    pure val DeliverPacketToProviderTimeoutTest_testState = DeliverPacketToProviderHappyPathTest_testState.with(
+            "providerState", DeliverPacketToProviderHappyPathTest_testState.providerState.with(
+                "chainState", DeliverPacketToProviderHappyPathTest_testState.providerState.chainState.with(
+                    // set the timestamp to be after the timeout
+                    "lastTimestamp", CcvTimeout.get("sender") + 1
+                )
+            )
+        )
+
+    pure val DeliverPacketToProviderTimeoutTest_resultAndTimeout = deliverPacketToProvider(DeliverPacketToProviderTimeoutTest_testState, "sender")
+
+    run DidTimeOut_DeliverPacketToProviderTimeoutTest =
+    {
+        val result = DeliverPacketToProviderTimeoutTest_resultAndTimeout._1
+        val timedOut = DeliverPacketToProviderTimeoutTest_resultAndTimeout._2
+        val newProviderState = result.newState.providerState
+        val newConsumerState = result.newState.consumerStates.get("sender")
+        not(result.hasError) and
+        timedOut
+    }
+
+    run StateModificationOK_DeliverPacketToProviderTimeoutTest =
+    {
+        val result = DeliverPacketToProviderTimeoutTest_resultAndTimeout._1
+        val timedOut = DeliverPacketToProviderTimeoutTest_resultAndTimeout._2
+        val newProviderState = result.newState.providerState
+        val newConsumerState = result.newState.consumerStates.get("sender")
+        not(result.hasError) and
+        newProviderState.receivedMaturations.size() == 0 and
+        newProviderState.consumerStatus.get("sender") == TIMEDOUT
+    }
+
+    run ConsumerStatusMapHappyPathTest = 
+    {
+        val currentConsumerStatusMap = Map(
+            "chain1" -> NOT_CONSUMER,
+            "chain2" -> RUNNING,
+            "chain3" -> STOPPED
+        )
+        val res = StartStopConsumers(
+            currentConsumerStatusMap,
+            Set("chain1"),
+            Set("chain2"),
+            Set()
+        )
+        res._2 == "" and
+        res._1.get("chain1") == RUNNING and
+        res._1.get("chain2") == STOPPED and
+        res._1.get("chain3") == STOPPED
+    }
+
+    run ConsumerStatusMapAlreadyRunningTest =
+    {
+        val currentConsumerStatusMap = Map(
+            "chain1" -> NOT_CONSUMER,
+            "chain2" -> RUNNING,
+            "chain3" -> STOPPED
+        )
+        val res = StartStopConsumers(
+            currentConsumerStatusMap,
+            Set("chain2"),
+            Set("chain3"),
+            Set()
+        )
+        res._2 == "cannot start a consumer that is stopped or already a consumer"
+    }
+
+    run ConsumerStatusMapAlreadyStoppedTest =
+    {
+        val currentConsumerStatusMap = Map(
+            "chain1" -> NOT_CONSUMER,
+            "chain2" -> RUNNING,
+            "chain3" -> STOPPED
+        )
+        val res = StartStopConsumers(
+            currentConsumerStatusMap,
+            Set("chain1"),
```

Comment:
Do we need this here? We could have `Set()` since we do not check anything for `chain1` in this specific test?

Reply:
Yeah, true, but I think this doesn't really matter. Unless you think this makes it much harder to understand